modules: tf-m: split configuration into multiple Kconfig files

ioannisg opened the pull request:
Split the Kconfig-based TF-M configuration into multiple Kconfig files, for readability and to be able to easily override whole Kconfig files in downstream projects.

hakonfam committed:
modules: tf-m: split TF-M configuration in multiple Kconfig files

The TF-M project is higly configurable and we end up having
a growing number of Kconfig option definitions in the main
Kconfig file. This commit splits out the configuration options
for the TF-M partition and for the crypto partition configuration
into separate Kconfig files, for ease of reading. The commit
does not introduce behavioral changes.

Signed-off-by: Håkon Øye Amundsen <haakon.amundsen@nordicsemi.no>
Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

hakonfam commented on modules/trusted-firmware-m/Kconfig:
```diff
@@ -1,377 +1,12 @@
-		Non-Secure application firmware.
-
-endif # BUILD_WITH_TFM
+rsource "Kconfig.tfm.partitions"
```
Both these files are "covered" internally by `if BUILD_WITH_TFM` - is it still necessary to have that check here?

ioannisg replied:
Removed, @hakonfam , thanks

ioannisg committed:
modules: tfm: remove redundant if BUILD_WITH_TFM guard in Kconfig

We do not need the if BUILD_WITH_TFM guard in the main
Kconfig file, as the guards are present insude the respective
Kconfig files that are sourced by the main Kconfig.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

hakonfam reviewed (approved):
My initial commit, so a bit biased :)

ioannisg commented:
@microbuilder @oyvindronningstad @lairdjm, quick look?

oyvindronningstad commented:
Since we lost a change downstream, I just want to ask if this we are sure there is unintended change?

hakonfam commented:
> Since we lost a change downstream, I just want to ask if this we are sure there is unintended change?

Should not be a problem since its just a duplicate check, I guess its enough that it will be resolved on the next upmerge.

lairdjm commented on modules/trusted-firmware-m/Kconfig.tfm:
```diff
@@ -0,0 +1,242 @@
+	  When this is 1, the S and NS are considered as 1 image and must be
+	  updated in one atomic operation. When this is 2, they are split and
+	  can be updated independently if dependency requirements are met.
+
+
```
Excess space
